modules/docs/arrow-docs/docs/docs/arrow-fx/integrations/kotlinxcoroutines/README.md

```diff
@@ -0,0 +1,170 @@
+---
+layout: docs-core
+title: kotlinx.coroutines
+permalink: /docs/integrations/kotlinxcoroutines/
+---
+
+*Note* that as of Arrow 0.9.0, we have deprecated support for `Deferred` from `kotlinx.coroutines`. Using `Deferred` as a return type is considered a smell by the library owners (see [here](https://github.com/Kotlin/kotlinx.coroutines/blob/master/docs/composing-suspending-functions.md#async-style-functions) and [here](http://youtrack.jetbrains.com/issue/KT-25620)), and we were not able to make it work consistently against some of the Laws. 
+
+But we have not given up support for suspend functions! If you would like to use `suspend fun`, you can do so using the `arrow-fx` and this module.
+
+# Kotlin Coroutines and runtime support
```

Comment:
Kotlin offer's a `suspend` system in the language, and it offers intrinsics in the standard library to build a library on top. These `intrinsic` functions allow you to `startCoroutine`s, `suspendCoroutine`s, build `CoroutineContext`s and so on.

Kotlin's language suspension support can be found in the [kotlin.coroutines](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.coroutines/index.html) package.

There are currently two libraries that provide a runtime for the language's suspension system.

- [Arrow Fx](https://arrow-kt.io/docs/fx/)
- [KotlinX Coroutines](https://github.com/Kotlin/kotlinx.coroutines)

They can easily interop with each-other and Arrow Fx's integration module offers certain combinators to use Arrow Fx's with KotlinX structured concurrency in frameworks that have chosen to incorporate the KotlinX Coroutines library such as Android and Ktor.



It's a suggestion to replace `Kotlin Coroutines and runtime support`.
I found reading the current `Kotlin Coroutines and runtime support` a bit confusing, and less explicitly indicates the difference between language and library.